react/components/Collapsible/index.js

```diff
@@ -46,6 +57,27 @@ class Collapsible extends Component {
     this.setState({
       height: childrenHeight,
     })
+    this.resetHeight()
+  }
+
+  closeCard = () => {
+    const childrenHeight = this.childrenRef.current.offsetHeight
+    this.childrenRef.current.style.height = childrenHeight
+    /** after force setting element height like the line above
+     * you have to force layout / reflow so the height value
+     * may actually apply. You can do this by requesting
+     * element offsetHeight again, like the line below
+     */
+    this.childrenRef.current.offsetHeight
+    this.setState(
+      {
+        height: childrenHeight,
+      },
+      () => {
+        clearTimeout(this.closeTimeout)
+        this.closeTimeout = setTimeout(() => this.setState({ height: 0 }), 0)
```

Comment:
Why are you using `setTimeout` here? Wouldn't be better use `requestAnimationFrame` ?